### PR TITLE
add org property to role list if exists

### LINKED ISSUE #38: Add extra properties to `roles.list()` API response. Specifically...
```json
{
  "org": {
      "id": "5b969dc9901e2af192780a34",
      "name": "foo"
    }
}
```

### PR DESCRIPTION
**Pull request by spideynr**
fixes #38

@jmike I had a problem while testing the Roles list. I am expecting that Roles list API should also include global roles the user has access to. Is that not the case? Do global rules not include the `yeep.role.read` domain?

Apparently not. Should we? The test is currently failing and i also added some precautions on the query fetch for the case an org does not exist for a given role so that it doesn't crash in the future

### CODE REVIEW

**Comment by jmike** [comment]
@spideynr I understand the issue, but this needs further discussion. I am not sure the user should have read access to roles they 've been assigned with. It might be an unnecessary complexity. Can you think of an example this would be useful?

We could take this offline to save time and reconvene here with the outcome of our discussion.

**Comment by spideynr** [comment]
> @spideynr I understand the issue, but this needs further discussion. I am not sure the user should have read access to roles they 've been assigned with. It might be an unnecessary complexity. Can you think of an example this would be useful?
> 
> We could take this offline to save time and reconvene here with the outcome of our discussion.

@jmike i inverted the test. The PR is rebased with master and ready to close when you are

File: server/api/role/list/service.js [hunk 1]
```diff
@@ -60,6 +61,20 @@ async function listRoles(db, { q, limit, cursor, scopes }) {
...
+        from: 'orgs',
+        localField: 'scope',
+        foreignField: '_id',
+        as: 'org',
+      },
+    },
+    {
+      $unwind: {
+        path: '$org',
+          'preserveNullAndEmptyArrays': true,
```
Comment (jmike) [inline server/api/role/list/service.js]: @spideynr can you please fix the indentation here?

File: server/api/role/list/service.js [hunk 2]
```diff
@@ -68,6 +83,10 @@ async function listRoles(db, { q, limit, cursor, scopes }) {
   return roles.map((role) => ({
     id: role._id.toHexString(),
     name: role.name,
+    org: {
+      id: get(role, 'org._id') && get(role, 'org._id').toHexString(),
```
Comment (jmike) [inline server/api/role/list/service.js]: I suggest to change this as follows...

```
org: has(role, ['org', '_id']) ? {
  id: role.org._id.toHexString(),
  name: role.org.name,
} : null
```

Here's my reasoning:

1. If `org._id` doesn't exist then the `org` property should be null (as we discussed on slack);
2. You needn't check for `org.name` and `org._id` separately. If `org._id` exists then there is definitely an `org.name` as well.

File: server/api/role/list/controller.test.js [hunk 3]
```diff
@@ -97,6 +112,12 @@ describe('api/v1/role.list', () => {
       .send();
 
     expect(res.status).toBe(200);
+    // NOTE: Even though the user is assigned 3 roles, one of them is global.
```
Comment (jmike) [inline server/api/role/list/controller.test.js]: This comment here doesn't make much sense to me as an external reader. It should be either simplified or removed altogether imo.

Comment (spideynr) [reply server/api/role/list/controller.test.js]: eh, i seem to be really bad at explaining technical stuff over text /shrug .

**Pull request merged**